File: seed/csharp-sdk/trace/src/SeedTrace/Submission/Types/RecordingResponseNotification.cs
using System.Text.Json.Serialization;
using SeedTrace;

#nullable enable

namespace SeedTrace;

public record RecordingResponseNotification
{
    [JsonPropertyName("submissionId")]
    public required string SubmissionId { get; }

    [JsonPropertyName("testCaseId")]
    public string? TestCaseId { get; }

    [JsonPropertyName("lineNumber")]
    public required int LineNumber { get; }

    [JsonPropertyName("lightweightStackInfo")]
    public required LightweightStackframeInformation LightweightStackInfo { get; }

    [JsonPropertyName("tracedFile")]
    public TracedFile? TracedFile { get; }
}

File: seed/php-model/trace/src/Submission/RecordingResponseNotification.php
<?php

namespace Seed\Submission;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;

class RecordingResponseNotification extends SerializableType
{
    /**
     * @var string $submissionId
     */
    #[JsonProperty("submissionId")]
    public string $submissionId;

    /**
     * @var ?string $testCaseId
     */
    #[JsonProperty("testCaseId")]
    public ?string $testCaseId;

    /**
     * @var int $lineNumber
     */
    #[JsonProperty("lineNumber")]
    public int $lineNumber;

    /**
     * @var LightweightStackframeInformation $lightweightStackInfo
     */
    #[JsonProperty("lightweightStackInfo")]
    public LightweightStackframeInformation $lightweightStackInfo;

    /**
     * @var ?TracedFile $tracedFile
     */
    #[JsonProperty("tracedFile")]
    public ?TracedFile $tracedFile;

    /**
     * @param array{
     *   submissionId: string,
     *   testCaseId?: ?string,
     *   lineNumber: int,
     *   lightweightStackInfo: LightweightStackframeInformation,
     *   tracedFile?: ?TracedFile,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->submissionId = $values['submissionId'];
        $this->testCaseId = $values['testCaseId'] ?? null;
        $this->lineNumber = $values['lineNumber'];
        $this->lightweightStackInfo = $values['lightweightStackInfo'];
        $this->tracedFile = $values['tracedFile'] ?? null;
    }
}
